src/figuresClasses.ts

```
@@ -1,11 +1,95 @@
-export interface Figure {}
+type ShapeType = 'triangle' | 'circle' | 'rectangle';
+type ColorType = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: ShapeType;
+  color: ColorType;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: ShapeType = 'triangle';
+
+  color: ColorType;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: ColorType, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be positive numbers.');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('These sides cannot form a triangle.');
+    }
+
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: ShapeType = 'circle';
+
+  color: ColorType;
 
-export class Circle implements Figure {}
+  radius: number;
 
-export class Rectangle implements Figure {}
+  constructor(color: ColorType, radius: number) {
+    if (radius <= 0) {
+      throw new Error('Radius must be a positive number.');
```

This violates checklist item: 'Error messages thrown in constructors must clearly describe the specific validation failure (e.g., "Width must be greater than 0", "Sides 1, 2 and 3 can't form a triangle")'.

The error message 'Radius must be a positive number.' is not specific about the parameter name and value. You should throw an error like 'Circle radius must be greater than 0, got {radius}'.